src/main/java/bio/terra/service/policy/PolicyService.java

```diff
@@ -76,6 +78,8 @@ public void createPao(
 
   public void updatePao(TpsPaoUpdateRequest updateRequest, UUID resourceId) {
     TpsApi tpsApi = policyApiService.getPolicyApi();
+    // Setting the update mode is required, but not enforced by the terra-policy-client.
```

Comment:
I was hoping that by updating the client, we might get this enforcement, but no such luck :(

Reply:
Do you mean TPS doesn't enforce that the update mode is set even though it is required? 🤔

Reply:
Their client doesn't check that update mode is set, so when our request makes it to the API, it fails there.  The client only checks that a request body is provided:
```
            public TpsPaoUpdateResult updatePao(TpsPaoUpdateRequest body, UUID objectId) throws ApiException {
            Object localVarPostBody = body;
                    // verify the required parameter 'body' is set
                    if (body == null) {
                    throw new ApiException(400, "Missing the required parameter 'body' when calling updatePao");
                    }
                    // verify the required parameter 'objectId' is set
                    if (objectId == null) {
                    throw new ApiException(400, "Missing the required parameter 'objectId' when calling updatePao");
                    }
```

I thought that maybe their autogenned `TpsPaoUpdateRequest` object would annotate `updateMode` as required, I've seen that in TDR's autogenned resources with required fields, but it didn't.

```
@Schema(description = "Update request body")
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2023-04-24T16:24:35.491427893Z[Etc/UTC]")
public class TpsPaoUpdateRequest {
  @JsonProperty("removeAttributes")
  private TpsPolicyInputs removeAttributes = null;

  @JsonProperty("addAttributes")
  private TpsPolicyInputs addAttributes = null;

  @JsonProperty("updateMode")
  private TpsUpdateMode updateMode = null;
…
```

Reply:
Ah interesting. At some point, I'll have to walk y'all on the client generation stuff I did for DrsHub.  Could be a thing that the policy team is interested for making the client generate that kind of thing

Reply:
That would be super cool!